src/firewall/command.py

```diff
@@ -625,3 +627,210 @@ def get_ipset_entries_from_file(self, filename):
                 entries_set.add(line)
         f.close()
         return entries
+
+    def show_state_and_exit(self, fw, usage_text=None, verbose=False):
+        config = fw.config()
+
+        exit_code = 0
+        msg = ""
+        state = fw.get_state()
+        if state == "RUNNING":
+            s = "running"
+        elif state == "FAILED":
+            s = "failed"
+        elif state == "NOT_AUTHORIZED":
+            s = "not authorized"
+            exit_code = errors.NOT_AUTHORIZED
+        else:
+            state = "NOT_RUNNING"
+            s = "not running"
+            exit_code = errors.NOT_RUNNING
+        if verbose or state != 'RUNNING':
+            msg += f"State: {s}\n"
+
+        try:
+            v = fw.real_get_property("version", dbus.String)
```

Comment:
If `get_state()` above returned `NOT_RUNNING` then I don't follow why we would attempt any other DBUS call. IMO we should only implement the `real_get_property()` function for the state. If firewalld is not running, then we're not going to have any useful information to print.

Reply:
The code after NOT_RUNNING anyway handles D-Bus exceptions for every "raw_()" request (because there could be other errors, or firewalld quit just in that moment). As it's already handled, adding more special case seems unnecessary.

In general, since we make a series of D-Bus requests, the configuration in the server can change at any time (in between those requests). So with the current approach, it's anyway not possible to get a 100% self-consistent output. E.g. you might see "State: running" followed by "Version: unknown".

Not to touch anything after NOT_RUNNING, doesn't solve this problem in general, so it's not clear what that the special handling solves.

(btw, libnm's NMClient builds a client-side, fully-consistent cache. Then you can show the consistent information from one point in time. But that is hard to do).